static/app/views/profiling/landing/chart.tsx

```diff
@@ -0,0 +1,184 @@
+import {useMemo} from 'react';
+import {browserHistory, withRouter, WithRouterProps} from 'react-router';
+import {useTheme} from '@emotion/react';
+import {Location} from 'history';
+
+import ChartZoom from 'sentry/components/charts/chartZoom';
+import OptionSelector from 'sentry/components/charts/optionSelector';
+import ScatterChart from 'sentry/components/charts/scatterChart';
+import {
+  ChartContainer,
+  ChartControls,
+  InlineContainer,
+} from 'sentry/components/charts/styles';
+import TransitionChart from 'sentry/components/charts/transitionChart';
+import TransparentLoadingMask from 'sentry/components/charts/transparentLoadingMask';
+import {getSeriesSelection} from 'sentry/components/charts/utils';
+import {Panel} from 'sentry/components/panels';
+import Placeholder from 'sentry/components/placeholder';
+import {t} from 'sentry/locale';
+import {SelectValue} from 'sentry/types';
+import {EChartClickHandler, SeriesDataUnit} from 'sentry/types/echarts';
+import {Trace} from 'sentry/types/profiling/trace';
+import {defined} from 'sentry/utils';
+import {axisLabelFormatter, tooltipFormatter} from 'sentry/utils/discover/charts';
+import getDynamicText from 'sentry/utils/getDynamicText';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface Props extends WithRouterProps {
+  location: Location;
+  traces: Trace[];
+  end?: string;
```

Comment:
I think @evanpurkhiser did some work recently and we have a type for these period props - maybe we can just extend that

Reply:
I think you're referring to the work on page filters the core UI team is working on. It's not clear if those APIs are stable yet as it's actively being worked on and I don't see a type for just `start/end/period` (maybe I missed it). So I think we'll leave this alone for now and revisit again in the future.